### PR TITLE
Add a Parameter to Configure Worker count for fleet-controller

### PR DESCRIPTION
**Pull request by Tommy12789**
Refers to https://github.com/rancher/fleet/issues/2415

Added a new environment var for each type of reconciler (gitrepo, bundle, bundledeployment) to configure their worker count (in values.yaml)

No test for this, we need a large scale fleet to test to test it

### CODE REVIEW

File: charts/fleet/templates/deployment.yaml [hunk 1]
```diff
@@ -64,6 +64,18 @@ spec:
         - name: CATTLE_DEV_MODE
           value: "true"
         {{- end }}
+        {{- if $.Values.controller.reconciler.workers.gitrepo }}
+        - name: CONTROLLER_GITREPO_RECONCILER_WORKERS
```
Comment (manno) [inline charts/fleet/templates/deployment.yaml:68]:
```suggestion
        - name: GITREPO_RECONCILER_WORKERS
```

I'd remove the controller prefix on all envs, for brevity.

File: charts/fleet/values.yaml [hunk 2]
```diff
@@ -90,3 +90,12 @@ leaderElection:
...
   renewDeadline: 25s
+
+## Fleet controller configuration
+controller:
+  reconciler:
+    # The number of workers that are allowed to each type of reconciler
+    workers:
+      gitrepo: "1"
+      bundle: "2"
+      bundledeployment: "3"
```
Comment (manno) [inline charts/fleet/values.yaml:101]:
```suggestion
      bundledeployment: "1"
```

I'd default them all to 1, I think that's the upstream default, too.

Comment (Tommy12789) [reply charts/fleet/values.yaml:101]: Of course, I forgot to reset them !

File: charts/fleet/templates/deployment.yaml (hunk 1 above)
Comment (Tommy12789) [reply charts/fleet/templates/deployment.yaml]: Just changed this TY Mario !

File: internal/cmd/controller/root.go [hunk 3]
```diff
@@ -54,6 +54,12 @@ type LeaderElectionOptions struct {
 	RetryPeriod *time.Duration
 }
 
+type ControllerReconcilerWorkers struct {
+	Gitrepo          int
```
Comment (manno) [inline internal/cmd/controller/root.go:58]: I just noticed it's GitRepo everywhere else

```suggestion
	GitRepo          int
```

File: internal/cmd/controller/root.go [hunk 4]
```diff
@@ -54,6 +54,12 @@ type LeaderElectionOptions struct {
...
+	Bundle           int
+	Bundledeployment int
```
Comment (manno) [inline internal/cmd/controller/root.go:60]:
```suggestion
	BundleDeployment int
```

**Review by manno — CHANGES_REQUESTED** [review]

File: internal/cmd/controller/root.go (hunk 4 above)
Comment (Tommy12789) [reply internal/cmd/controller/root.go]: changed !

File: internal/cmd/controller/root.go (hunk 3 above)
Comment (Tommy12789) [reply internal/cmd/controller/root.go]: changed !

**Review by manno — APPROVED** [review]

**Pull request merged**